neo/Consensus/ConsensusService.cs

```diff
@@ -268,11 +272,29 @@ private void OnConsensusPayload(ConsensusPayload payload)
         private void OnPersistCompleted(Block block)
         {
             Log($"persist block: {block.Hash}");
-            block_received_time = TimeProvider.Current.UtcNow;
+            reference_block_time = TimeProvider.Current.UtcNow;        
+            theoreticalDelay = calcNetworkDelayOrClockDriftWithLimits(reference_block_time.ToTimestamp(), block.Timestamp);
+            reference_block_time.AddSeconds((double)-theoreticalDelay);
             knownHashes.Clear();
             InitializeConsensus(0);
         }
 
+        private uint calcNetworkDelayOrClockDriftWithLimits(uint localClockTimestamp, uint lastBlockTimestamp)
+        {
+            if (localClockTimestamp > lastBlockTimestamp)
```

Comment:
Don’t call function unless this condition is met?

Reply:
Alternatively at least invert the if for less nesting.

Reply:
If you keep the if inside you should probably make it return true or false, and use an out variable for the time adjustment. That way only if it returns true you add the offset.

Reply:
In the current way it always add 0 is it is equal, otherwise adds the offset.

Reply:
Which way to you think is better and faster?

Reply:
Check this now @jsolman